auth/refresh_token.go

```diff
@@ -13,38 +13,115 @@ import (
 	goahttp "goa.design/goa/v3/http"
 )
 
-// refreshTokenSource contains the logic to create new session using OAuth tokens
-type refreshTokenSource struct {
-	endpoint     *url.URL
-	refreshToken string
-	accessToken  string
-	refreshAt    time.Time
-	doer         goahttp.Doer
-}
+type (
+	// flexeraRefreshTokenSource contains the logic to create new sessions using OAuth2 tokens
+	flexeraRefreshTokenSource struct {
+		endpoint     *url.URL
+		refreshToken string
+		accessToken  string
+		refreshAt    time.Time
+		doer         goahttp.Doer
+	}
 
-// Grant from CM API 1.5
-type Grant struct {
-	ExpiresIn   int    `json:"expires_in"`
-	AccessToken string `json:"access_token"`
-}
+	// rsRefreshTokenSource contains the logic to create new session using OAuth tokens using the legacy RightScale login system
+	rsRefreshTokenSource struct {
+		endpoint     *url.URL
+		refreshToken string
+		accessToken  string
+		refreshAt    time.Time
+		doer         goahttp.Doer
+	}
+
+	// Grant represents the grant response returned by the token endpoint
+	Grant struct {
+		ExpiresIn   int    `json:"expires_in"`
+		AccessToken string `json:"access_token"`
+	}
+)
+
+var (
+	flexeraHostRegexp = regexp.MustCompile(`^login\.flexera(?:test)?\.com$`)
```

Comment:
Would it make sense to add `login.flexera.eu` to the list here since we'll soon have policies there?

Reply:
It would, but I didn't do it yet as I don't know what the other EU host names for Policies will look like yet.